app/services/katello/rhsm_fact_parser.rb

```diff
@@ -55,7 +55,13 @@ def operatingsystem
       return nil if name.nil? || version.nil?
 
       os_name = distribution_to_puppet_os(name)
-      major, minor = version.split('.')
+      version_splits = version.split('.')
```

Comment:
It should be noted you can pass a parameter to `.split` about how often it is allowed to split. So:
```
[1] pry(main)> '22.04.2'.split('.', 3)
=> ["22", "04", "2"]
[2] pry(main)> '7.8'.split('.', 3)
=> ["7", "8"]
```

Perhaps it should be:
```ruby
if name.match?(/ubuntu/i)
  x, y, minor = version.split('.', 3)
  major = "#{x}.#{y}"
else
  major, minor = version.split('.', 2)
end
```

Reply:
Thanks for the improvement. I also now rely on the normalized OS name and applied your suggestion